### PR TITLE
Improve searching

### LINKED ISSUE #110: Move search filters out of their select boxes
**Describe the solution you'd like**
image

**Additional context**
Something like radio buttons would be nicer.

### PR DESCRIPTION
**Pull request by NicoAiko**
+ Search button
+ Loading
+ Error message when failed
+ Filter component
+ Expansion panel for mobile
@TODO: Refactor whole searching

Fixes #110

### CODE REVIEW

File: src/assets/locales/fr.json [hunk 1]
```diff
@@ -134,9 +134,11 @@
     },
     "search": {
       "searchQuery": "Chercher...",
+      "searchFilter": "Filtre de Recherche",
```
Comment (TheRaphael0000) [inline src/assets/locales/fr.json]: The capital R isn't required. The other add is ok. :D

File: src/assets/locales/id.json [hunk 2]
```diff
@@ -134,9 +134,11 @@
     },
     "search": {
       "searchQuery": "Cari...",
+      "searchFilter": "Search filter",
```
Comment (nattadasu) [inline src/assets/locales/id.json]:
```suggestion
      "searchFilter": "Penyaringan pencarian",
```

File: src/assets/locales/id.json [hunk 3]
```diff
@@ -134,9 +134,11 @@
...
       "adultContent": {
         "label": "Konten dewasa",
         "hint": "Membantu mengelola pencarian untuk konten dewasa maupun bukan",
+        "both": "No filter",
```
Comment (nattadasu) [inline src/assets/locales/id.json]:
```suggestion
        "both": "Tanpa penyaringan",
```

File: src/assets/locales/ms.json [hunk 4]
```diff
@@ -134,9 +134,11 @@
     },
     "search": {
       "searchQuery": "Cari...",
+      "searchFilter": "Search filter",
       "adultContent": {
         "label": "Kandungan dewasa",
         "hint": "Membantu menguruskan carian untuk kandungan dewasa dan bukan dewasa",
+        "both": "No filter",
```
Comment (nattadasu) [inline src/assets/locales/ms.json]:
```suggestion
        "both": "Tiada penapisan",
```

File: src/assets/locales/ms.json [hunk 5]
```diff
@@ -134,9 +134,11 @@
     },
     "search": {
       "searchQuery": "Cari...",
+      "searchFilter": "Search filter",
```
Comment (nattadasu) [inline src/assets/locales/ms.json]:
```suggestion
      "searchFilter": "Penapis carian",
```

File: src/assets/locales/ar.json [hunk 6]
```diff
@@ -134,9 +134,11 @@
     },
     "search": {
       "searchQuery": "بحث...",
+      "searchFilter": "Search filter",
```
Comment (nattadasu) [inline src/assets/locales/ar.json]:
```suggestion
      "searchFilter": "تصفية البحث",
```

File: src/assets/locales/ar.json [hunk 7]
```diff
@@ -134,9 +134,11 @@
...
       "adultContent": {
         "label": "محتوى للبالغين",
         "hint": "يساعد على إدارة البحث للعثور على البالغين المحتوى وغير البالغين المحتوى",
+        "both": "No filter",
```
Comment (nattadasu) [inline src/assets/locales/ar.json]:
```suggestion
        "both": "لا تصفية",
```

**Comment by NicoAiko** [comment]
- [x] German

- [x] English

- [x] Japanese

- [x] French

- [ ] Simplified Chinese

- [ ] Brazilian Portuguese

- [x] Arabic

- [x] Indonesian

- [x] Malay

File: src/assets/locales/zh_CN.json [hunk 8]
```diff
@@ -134,9 +134,11 @@
     },
     "search": {
       "searchQuery": "搜索……",
+      "searchFilter": "Search filter",
```
Comment (Mornwind) [inline src/assets/locales/zh_CN.json]:
```suggestion
      "searchFilter": "筛选设置",
```

File: src/assets/locales/zh_CN.json [hunk 9]
```diff
@@ -134,9 +134,11 @@
...
       "adultContent": {
         "label": "R-18 内容",
         "hint": "留空以同时搜索全年龄内容和 R-18 内容",
+        "both": "No filter",
```
Comment (Mornwind) [inline src/assets/locales/zh_CN.json]:
```suggestion
        "both": "未筛选",
```

File: src/assets/locales/pt_BR.json [hunk 10]
```diff
@@ -134,9 +134,11 @@
     },
     "search": {
       "searchQuery": "Pesquisar...",
+      "searchFilter": "Search filter",
```
Comment (frozaum) [inline src/assets/locales/pt_BR.json]: Filtro de pesquisa

File: src/assets/locales/pt_BR.json [hunk 11]
```diff
@@ -134,9 +134,11 @@
...
       "adultContent": {
         "label": "Conteúdo Adulto",
         "hint": "Mantenha-se livre para pesquisar conteúdo adulto e não-adulto",
+        "both": "No filter",
```
Comment (frozaum) [inline src/assets/locales/pt_BR.json]: Sem filtro

**Pull request merged**